enhancements/security/security-profiles-operator.md

```diff
@@ -0,0 +1,627 @@
+been developed. The intent is that selinuxd allows one to simply drop a file
+on a pre-determined directory to install a SELinux policy, and that policy will
+get picked up and installed by the daemon. Thus, the SPOD instance simply
+places the policy into a directory, the selinuxd instance picks it up
+and handles the installation and tracking of policies.
+
+A `SelinuxProfile` object looks as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: SelinuxProfile
+metadata:
+  name: errorlogger
+  namespace: my-namespace
+spec:
+  policy: |
+    (blockinherit container)
+    (allow process var_log_t ( dir ( open read getattr lock search ioctl add_name remove_name write ))) 
+    (allow process var_log_t ( file ( getattr read write append ioctl lock map open create  ))) 
+    (allow process var_log_t ( sock_file ( getattr read write append open  )))
+```
+
+It is mostly a wrapper around [SELinux's CIL language](https://github.com/SELinuxProject/cil)
+with the main block and naming handled by the operator. Naming of policies
+as implemented by the daemon is as follows `<object name>_<object namespace>`. Thus,
+we're able to distinguish between policies installed in different namespaces.
+
+To ease usability and allow for developers to easily bind profiles to workloads,
+an object called `ProfileBinding` was created. The intent is for the binding to
+declare what profile goes to a ceratain image. The binding is then enforced by a
+webhook which mutates the pod definition to add the appropriate profile
+to its security context.
+
+The `ProfileBinding` object looks as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: ProfileBinding
+metadata:
+  name: profile-binding
+spec:
+  profileRef:
+    kind: SeccompProfile
+    name: profile-allow-unsafe
+  image: nginx:1.19.1
+```
+
+This would help manage security profiles and match them to workloads
+accross the application's lifecycle. Where one could upgrade their
+workload, and create a binding that matches the new application's image
+tag.
+
+Finally, to aide developers in writing profiles, there is a `ProfileRecording`
+object. The object would look as follows:
+
+```yaml
+---
+apiVersion: security-profiles-operator.x-k8s.io/v1alpha1
+kind: ProfileRecording
+metadata:
+  # The name of the Recording is the same as the resulting `SeccompProfile` CRD
+  # after reconciliation.
+  name: test-recording
+spec:
+  kind: SeccompProfile
+  podSelector:
+    matchLabels:
+      app: alpine
+```
+
+The operator automates the recording of profiles, and, when the
+workload is done, it's able to output a ready-to-use profile
+with the privileges that the application needs.
+
+The `SeccompProfile` objects rely on the [OCI Seccomp BPF Hook](https://github.com/containers/oci-seccomp-bpf-hook)
+to do the actual recording. So we'd need to ensure this hook is available in RHCOS
+before enabling this feature.
```

Comment:
Let's mention that we need CRI-O v1.21.0 for this to work properly beside the OCI seccomp BPF hook. Let's also mention that the operator feature will not work in case both conditions are not met, but this has no influence on the overall functionality of the operator. This means the operator can also be used on previous OpenShift versions.

Reply:
@saschagrunert the CRI-O version depends on the OpenShift version. The intent is to target this towards OpenShift 4.10 and not any older version. CRI-O 1.21 IIRC is available in 4.9, so we're good on that side.